core/src/com/unciv/ui/worldscreen/unit/UnitActions.kt

```diff
@@ -482,6 +495,55 @@ object UnitActions {
         )
     }
 
+    private fun getRepairTurns(unit: MapUnit): Int {
+        val tile = unit.currentTile
+        if (!tile.isPillaged()) return 0
+        var repairTurns = tile.ruleset.tileImprovements["Repair"]!!.getTurnsToBuild(unit.civInfo, unit)
+        if (tile.improvement != null && tile.improvementIsPillaged) {
+            if (tile.getTileImprovement()!!.getTurnsToBuild(unit.civInfo, unit) < repairTurns)
+                repairTurns = tile.getTileImprovement()!!.getTurnsToBuild(unit.civInfo, unit)
+        } else {
+            if (tile.ruleset.tileImprovements[tile.roadStatus.name]!!.getTurnsToBuild(unit.civInfo, unit) < repairTurns)
+                repairTurns = tile.ruleset.tileImprovements[tile.roadStatus.name]!!.getTurnsToBuild(unit.civInfo, unit)
+        }
+        return repairTurns
+    }
+
+    private fun addRepairAction(unit: MapUnit, actionList: ArrayList<UnitAction>) {
+        if (!unit.hasUniqueToBuildImprovements) return
+        if (unit.isEmbarked()) return
+        val tile = unit.getTile()
+        if (!tile.isPillaged()) return
+
+        val couldConstruct = unit.currentMovement > 0
+                && !tile.isCityCenter()
+
+        val turnsToBuild = if (tile.improvementInProgress == UnitActionType.Repair.name) tile.turnsToImprovement
+        else getRepairTurns(unit)
+
+        actionList += UnitAction(UnitActionType.Repair,
+            title = "${UnitActionType.Repair} - [${turnsToBuild}${Fonts.turn}]",
+            action = getRepairAction(unit).takeIf { couldConstruct }
+        )
+    }
+
+    fun getRepairAction(unit: MapUnit): () -> Unit {
+        return {
+            unit.currentMovement = 0f
+            val tile = unit.currentTile
+            if (getRepairTurns(unit) - 1 == 0) {  // handle instant fix
```

Comment:
??? Why this weird phrasing? What's wrong with "if (x == -1)"? Should be equivalent

Reply:
Cleaned up